src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -0,0 +1,479 @@
+        iconItem = parentItem.child(row, 0)
+        curve = fileItem.data(qt.Qt.UserRole)
+        if curve is None:
+            return
+
+        legendIcon = LegendIcon(None, curve)
+        widget = qt.QWidget()
+        layout = qt.QHBoxLayout(widget)
+        layout.addWidget(legendIcon)
+        layout.setContentsMargins(4, 0, 4, 0)
+        self.setIndexWidget(iconItem.index(), widget)
+
+    def _createRemoveButton(self, row, parentItem):
+        if parentItem is None:
+            parentItem = self.model().getXParent()
+            index = self.model().index(0, 2)
+            button = qt.QToolButton(self)
+            button.setIcon(icons.getQIcon("remove"))
+            button.clicked.connect(
+                functools.partial(self._removeFile, None, parentItem)
+            )
+            self.setIndexWidget(index, button)
+            return
+
+        removeItem = parentItem.child(row, 2)
+        if removeItem:
+            button = qt.QToolButton(self)
+            button.setIcon(icons.getQIcon("remove"))
+            button.clicked.connect(
+                functools.partial(self._removeFile, removeItem, parentItem)
+            )
+            self.setIndexWidget(removeItem.index(), button)
+
+    def _removeFile(self, removeItem, parentItem):
+        if removeItem is None:
+            row = 0
+            removeItem = self.model().getXParent()
+
+        if removeItem:
+            row = removeItem.row()
+
+            if parentItem is None:
+                parentItem = self.model().getXParent()
+
+            parentItem.removeRow(row)
+
+            if parentItem == self.model().getXParent():
+                self.model()._xDataset = None
+                self.model()._addXFile("")
+                index = self.model().index(0, 2)
+                self.setIndexWidget(index, None)
+                self.model()._updateYCurvesWithDefaultX()
+
+    def dragEnterEvent(self, event):
+        super().dragEnterEvent(event)
+        self.acceptDrop(event)
+
+    def dragMoveEvent(self, event):
+        dropIndex = self.indexAt(event.pos())
+        if dropIndex.isValid() and self.model().itemFromIndex(dropIndex).parent() is None:
+            self.setDropHighlight("X")
+        else:
+            self.setDropHighlight("Y")
+        event.acceptProposedAction()
+
+    def dragLeaveEvent(self, event):
+        super().dragLeaveEvent(event)
+        self.setDropHighlight(None)
+
+    def dropEvent(self, event):
+        super().dropEvent(event)
+        byteString = event.mimeData().data("application/x-silx-uri")
+        url = silx.io.url.DataUrl(byteString.data().decode("utf-8"))
+
+        dropIndex = self.indexAt(event.pos())
+        if dropIndex.isValid() and self.model().itemFromIndex(dropIndex).parent() is None:
+            self.setX(url)
+        else:
+            self.addY(url)
+
+        self.setDropHighlight(None)
+        event.acceptProposedAction()
+
+    def acceptDrop(self, event):
```

Comment:
As this is a function name that can be close to qt API (but not something inherited if I am correct) and not highly necessary to have public (if I am correct again) maybe it could be move to protected